functions/venue.js

```diff
@@ -118,6 +118,34 @@ const checkUserIsOwner = async (venueId, uid) => {
     });
 };
 
+const checkIfUserIsVoted = async (venueId, pollId, userId) => {
+  await admin
+    .firestore()
+    .collection("venues")
+    .doc(venueId)
+    .collection("chats")
+    .doc(pollId)
+    .get()
+    .then((doc) => {
+      if (!doc.exists) {
+        throw new HttpsError("not-found", `Poll ${pollId} does not exist`);
+      }
+      const poll = doc.data();
+      if (!poll.votes.map(({ userId }) => userId).includes(userId)) return;
```

Comment:
I’d need to look at this function in its entirety to be sure, but the method I described in previous reviews for securely voting using FirldValue.arrayUnion with the user’s id shouldn’t need this kind of logic at all. We can get the user’s if from their auth credentials so we probably don’t need to even pass it as a param. At the very least we should ensure that the param and their auth match if we still want to explicitly pass it.

Reply:
And for future note, @mike-lvov ’S suggestion of .some() here is more efficient as it will shortcut as soon as it finds a match (and returns a Boolean), whereas .map() will always process every element (and returns an array)

Reply:
@0xdevalias 
securely voting array contains not only the user's id but also **questionId** (what actually he has voted for),
so IMHO I need to `.map` the votes array and left the only userId and after that check if this new array contains userId.
Can't understand how .some could help me.

Reply:
`!poll.votes.map(({ userId }) => userId).includes(userId)`

can be written as

`!poll.votes.some(({ userId: checkUserId }) => checkUserId === userId)

Reply:
fixed in 0b2f14a1d4eb95897597cb8c465a2c2719fd408b